HabitLogger.Arashi256/Program.cs

```diff
@@ -0,0 +1,767 @@
+    creationCount += TableExists(cs, "current_habit");
+    creationCount += TableExists(cs, "habits");
+    creationCount += TableExists(cs, "habit_logs");
+    if (creationCount < 3 || forceReset)
+    {
+        Console.WriteLine("The database need to be initialised. Any existing data will be lost. If this is your first use, choose 'y'");
+        if (GetValidYesNoResponse("OK to proceed (Y/N)?") == 'N')
+        {
+            return false;
+        }
+        else
+        {
+            Console.WriteLine("Resetting the database...please wait.");
+            ExecuteQuery(cs, "DROP TABLE IF EXISTS current_habit");
+            ExecuteQuery(cs, "DROP TABLE IF EXISTS habits");
+            ExecuteQuery(cs, "DROP TABLE IF EXISTS habit_logs");
+            ExecuteQuery(cs, @"CREATE TABLE IF NOT EXISTS habit_logs (Id INTEGER PRIMARY KEY AUTOINCREMENT, Date_Logged TEXT, Quantity INTEGER, Type INTEGER)");
+            ExecuteQuery(cs, @"CREATE TABLE IF NOT EXISTS habits (Id INTEGER PRIMARY KEY AUTOINCREMENT, Name TEXT, Units TEXT)");
+            ExecuteQuery(cs, @"CREATE TABLE IF NOT EXISTS current_habit (Id INTEGER)");
+            if (GetValidYesNoResponse("This is a freshly created database. Do you wish to populate it with test data (Y/N)?") == 'Y')
+            {
+                Console.WriteLine("Seeding test data, this might take some time...please wait.");
+                SeedTestData(cs);
+                Console.WriteLine("\nTest data seeding is COMPLETE. Please see habit and habit log lists. No current habit status has been set.");
+            }
+        }
+    }
+    return true;
+}
+
+static void SeedTestData(string cs)
+{
+    int habitNumber = 3;
+    Random random = new Random();
+    string[] habitNameData = { "Drinking water", "Exercise", "Mindfulness" };
+    string[] habitUnitsData = { "glasses", "workouts", "sessions" };
+    string randomDate = string.Empty;
+    int randomAmount = 0;
+    Habit currentHabit = null;
+    for (int j = 0; j < habitNumber; j++)
+    {
+        Console.Write($"Inserting test habit {habitNameData[j]}...");
+        ExecuteQuery(cs, $"INSERT INTO habits(Name, Units) VALUES ('{habitNameData[j]}', '{habitUnitsData[j]}')");
+        Console.Write($"DONE.\nInserting log records for {habitNameData[j]}...\n");
+        currentHabit = GetHabitFromName(cs, $"{habitNameData[j]}");
+        for (int i = 0; i < 33; i++)
+        {
+            do
+            {
+                Console.WriteLine("Getting non-duplicate date...");
+                randomDate = GetRandomDate(2023, 2024);
+            } while (CheckDuplicateHabitLogDate(cs, randomDate, currentHabit.Id));
+            randomAmount = random.Next(1, 31);
+            Console.Write($"{i} - Inserting {randomDate} - {randomAmount} for habit [{currentHabit.Id}]{currentHabit.Name}...");
+            ExecuteQuery(cs, $"INSERT INTO habit_logs(Date_Logged, Quantity, Type) VALUES ('{randomDate}', {randomAmount}, {currentHabit.Id})");
```

Comment:
You don't have to fix this to get the project approved but, take a look at parameterised commands, to prevent [SQL Injection](https://www.w3schools.com/sql/sql_injection.asp).

Reply:
Understood. I will use them next project for the coding tracker.